file name encryption/AES-SIV-512-B64URL.md

```diff
@@ -0,0 +1,178 @@
+# File Name Encryption using AES-SIV-512 + Base64-URL-Encoding
+
+## Directory Metadata
+
+Every directory requires certain metadata that affects the file name encryption of its direct children:
+
+* the seed used to derive keys
+* the directory ID
+
+This data is immutable and therefore linked with a directory eternally, surviving renames/moves. This data is stored in a file called `dir.uvf`, which is stored in two places:
+1. Within the parent dir (except for root), where it serves a link to the child dir
+2. In the child dir itself (allowing disaster recovery without the parent)
+
+The exact file structure of `dir.uvf` will be discussed in more detail [below](#format-of-diruvf-and-symlinkuvf).
+
+### Directory Seed
+
+At creation time, a directory's seed is always the `latestSeed` as defined in the [vault metadata file](../vault%20metadata/README.md#sensitive-metadata). In case of the root directory this happens to also be the `initialSeed`.
```

Comment:
@overheadhunter what's the motivation for moving away from using the same `nameKey` eternally to using the `latestSeed` for new directories?

Reply:
this allows key rotation to also be effective for names of nodes added to such directories.

Reply:
Does key rotation for names improve security in any way? Is it worth the additional lookup overhead in directory traversal of a vault?

Reply:
> Does key rotation for names improve security in any relevant way?

If users pick a product for the ability to encrypt names, they surely don't want names to leak. So yes, this is relevant.

> Is it worth the additional lookup/memory overhead in directory traversal of a vault - is this not restrictive?

Imo we get this for free, since we have the different seeds for file contents already anyway. Whether protecting a piece of information is *worth* additional effort depends on whether the leaking name is `document.pdf` or `company acquisition offer by Google.pdf`. Only the user can tell.